File: generate-test-audio.js
#!/usr/bin/env node

/**
 * Generiert eine minimal gültige WAV-Datei mit 1 Sekunde Stille
 * zum Testen der Mistral API
 */

const fs = require('fs');

function generateWavFile(filename, durationSeconds = 1) {
  const sampleRate = 16000; // 16 kHz
  const numChannels = 1; // Mono
  const bitsPerSample = 16;
  const numSamples = sampleRate * durationSeconds;
  const dataSize = numSamples * numChannels * (bitsPerSample / 8);
  
  // WAV Header erstellen
  const buffer = Buffer.alloc(44 + dataSize);
  let offset = 0;
  
  // RIFF chunk
  buffer.write('RIFF', offset); offset += 4;
  buffer.writeUInt32LE(36 + dataSize, offset); offset += 4;
  buffer.write('WAVE', offset); offset += 4;
  
  // fmt sub-chunk
  buffer.write('fmt ', offset); offset += 4;
  buffer.writeUInt32LE(16, offset); offset += 4; // Subchunk1Size
  buffer.writeUInt16LE(1, offset); offset += 2; // AudioFormat (1 = PCM)
  buffer.writeUInt16LE(numChannels, offset); offset += 2; // NumChannels
  buffer.writeUInt32LE(sampleRate, offset); offset += 4; // SampleRate
  buffer.writeUInt32LE(sampleRate * numChannels * bitsPerSample / 8, offset); offset += 4; // ByteRate
  buffer.writeUInt16LE(numChannels * bitsPerSample / 8, offset); offset += 2; // BlockAlign
  buffer.writeUInt16LE(bitsPerSample, offset); offset += 2; // BitsPerSample
  
  // data sub-chunk
  buffer.write('data', offset); offset += 4;
  buffer.writeUInt32LE(dataSize, offset); offset += 4;
  
  // Audio-Daten (Stille = 0)
  // Für einen einfachen Test lassen wir es bei Stille
  // (alle Samples sind bereits 0 durch Buffer.alloc)
  
  fs.writeFileSync(filename, buffer);
  console.log(`✅ Generiert: ${filename}`);
  console.log(`   Dauer: ${durationSeconds}s`);
  console.log(`   Sample Rate: ${sampleRate} Hz`);
  console.log(`   Channels: ${numChannels} (Mono)`);
  console.log(`   Bits per Sample: ${bitsPerSample}`);
  console.log(`   Dateigröße: ${buffer.length} bytes (${(buffer.length / 1024).toFixed(2)} KB)`);
}

// Generiere test-audio.wav
generateWavFile('test-audio.wav', 3);

console.log('');
console.log('⚠️  Hinweis: Dies ist eine stille Audio-Datei zum Testen des API-Formats.');
console.log('   Mistral wird "keine Sprache" erkennen oder einen leeren Text zurückgeben.');
console.log('   Für echte Tests solltest du eine Audio-Datei mit gesprochenem Text verwenden.');

File: generate_test_audio.py
#!/usr/bin/env python3
"""
Generiert eine minimal gültige WAV-Datei zum Testen der Mistral API
"""

import struct
import sys

def generate_wav(filename='test-audio.wav', duration=3, sample_rate=16000):
    """Generiert eine WAV-Datei mit Stille"""
    
    num_channels = 1  # Mono
    bits_per_sample = 16
    num_samples = sample_rate * duration
    data_size = num_samples * num_channels * (bits_per_sample // 8)
    
    with open(filename, 'wb') as f:
        # RIFF header
        f.write(b'RIFF')
        f.write(struct.pack('<I', 36 + data_size))
        f.write(b'WAVE')
        
        # fmt subchunk
        f.write(b'fmt ')
        f.write(struct.pack('<I', 16))  # Subchunk size
        f.write(struct.pack('<H', 1))   # Audio format (PCM)
        f.write(struct.pack('<H', num_channels))
        f.write(struct.pack('<I', sample_rate))
        f.write(struct.pack('<I', sample_rate * num_channels * bits_per_sample // 8))
        f.write(struct.pack('<H', num_channels * bits_per_sample // 8))
        f.write(struct.pack('<H', bits_per_sample))
        
        # data subchunk
        f.write(b'data')
        f.write(struct.pack('<I', data_size))
        
        # Audio data (silence = zeros)
        f.write(b'\x00' * data_size)
    
    file_size = 44 + data_size
    print(f'✅ Generiert: {filename}')
    print(f'   Dauer: {duration}s')
    print(f'   Sample Rate: {sample_rate} Hz')
    print(f'   Channels: {num_channels} (Mono)')
    print(f'   Bits per Sample: {bits_per_sample}')
    print(f'   Dateigröße: {file_size} bytes ({file_size / 1024:.2f} KB)')
    print()
    print('⚠️  Hinweis: Dies ist eine stille Audio-Datei zum Testen des API-Formats.')
    print('   Mistral wird "keine Sprache" erkennen oder einen leeren Text zurückgeben.')

if __name__ == '__main__':
    generate_wav()
